src/hotspot/share/oops/compressedKlass.cpp

```diff
@@ -32,9 +32,30 @@
 address CompressedKlassPointers::_base = nullptr;
 int CompressedKlassPointers::_shift = 0;
 size_t CompressedKlassPointers::_range = 0;
+uint64_t CompressedKlassPointers::_combo = 0;
 
 #ifdef _LP64
 
+void CompressedKlassPointers::set_base_and_shift(address thebase, int theshift) {
+  assert(UseCompressedClassPointers, "Why are we here?");
+  assert(theshift == 0 || theshift == LogKlassAlignmentInBytes, "invalid shift for klass ptrs");
+  _base = thebase;
+  _shift = theshift;
+
+  // we keep a composite word, `_combo`, containing base+shift+UseCCP, to load
+  // all three information with a single 64-bit load.
+  const uint64_t base_i = (uint64_t)_base;
+  assert((base_i & ~mask_base) == 0, "Base not aligned?");
+  assert(_shift <= 63, "Sanity");
+
+  _combo = (uint64_t)_base | (uint64_t)_shift | (1 << bitpos_useccp);
+
+  // validate combo.
+  assert(base() == _base, "combo encoding");
+  assert(shift() == _shift, "combo encoding");
+  assert(use_compressed_class_pointers() == UseCompressedClassPointers, "combo encoding");
```

Comment:
Pre-condition assert means `UseCompressedClassPointers` is always `true` here, can simplify the assert.